lib/taskana-core/src/main/java/org/taskana/model/mappings/QueryMapper.java

```diff
@@ -0,0 +1,139 @@
+package org.taskana.model.mappings;
+
+import java.util.List;
+
+import org.apache.ibatis.annotations.One;
+import org.apache.ibatis.annotations.Result;
+import org.apache.ibatis.annotations.Results;
+import org.apache.ibatis.annotations.Select;
+import org.taskana.impl.persistence.ClassificationQueryImpl;
+import org.taskana.impl.persistence.ObjectReferenceQueryImpl;
+import org.taskana.impl.persistence.TaskQueryImpl;
+import org.taskana.model.Classification;
+import org.taskana.model.ObjectReference;
+import org.taskana.model.Task;
+
+/**
+ * This class provides a mapper for all queries.
+ */
+public interface QueryMapper {
+
+    String OBJECTREFERENCEMAPPER_FINDBYID = "org.taskana.model.mappings.ObjectReferenceMapper.findById";
+    String CLASSIFICATION_FINDBYID = "org.taskana.model.mappings.ClassificationMapper.findById";
+
+    @Select("<script>SELECT t.ID, t.TENANT_ID, t.CREATED, t.CLAIMED, t.COMPLETED, t.MODIFIED, t.PLANNED, t.DUE, t.NAME, t.DESCRIPTION, t.PRIORITY, t.STATE, t.CLASSIFICATION_ID, t.WORKBASKETID, t.OWNER, t.PRIMARY_OBJ_REF_ID, t.IS_READ, t.IS_TRANSFERRED, t.CUSTOM_1, t.CUSTOM_2, t.CUSTOM_3, t.CUSTOM_4, t.CUSTOM_5, t.CUSTOM_6, t.CUSTOM_7, t.CUSTOM_8, t.CUSTOM_9, t.CUSTOM_10 "
+            + "FROM TASK t "
+            // Joins if Classification or Object Reference Query is needed
+            + "<if test='classificationQuery != null'>LEFT OUTER JOIN CLASSIFICATION c on t.CLASSIFICATION_ID = c.ID</if> "
+            + "<if test='objectReferenceQuery != null'>LEFT OUTER JOIN OBJECT_REFERENCE o on t.PRIMARY_OBJ_REF_ID = o.ID</if> "
+            + "<where>"
+            + "<if test='tenantId != null'>t.TENANT_ID = #{tenantId}</if> "
+            + "<if test='name != null'>AND t.NAME IN(<foreach item='item' collection='name' separator=',' >#{item}</foreach>)</if> "
```

Comment:
Bekomme ich hier nicht einen Syntaxfehler mit dem AND, wenn ich keinen Tenant gesetzt habe?